src/main/java/com/google/api/codegen/discovery/transformer/java/JavaMethodToViewTransformer.java

```diff
@@ -0,0 +1,168 @@
+        new SampleTypeTable(
+            new JavaTypeTable(), new JavaSampleTypeNameConverter(sampleConfig.packagePrefix()));
+    JavaSampleNamer namer = new JavaSampleNamer();
+    SampleTransformerContext context =
+        SampleTransformerContext.create(sampleConfig, sampleTypeTable, namer, method.getName());
+    return createSampleView(context);
+  }
+
+  private SampleView createSampleView(SampleTransformerContext context) {
+    addStaticImports(context);
+    SampleConfig sampleConfig = context.getSampleConfig();
+    SampleTypeTable typeTable = context.getTypeTable();
+
+    return SampleView.newBuilder()
+        .templateFileName(TEMPLATE_FILENAME)
+        .outputPath(context.getMethodName() + ".frag.java")
+        .apiTitle(sampleConfig.apiTitle())
+        .apiName(sampleConfig.apiName())
+        .apiVersion(sampleConfig.apiVersion())
+        .body(createSampleBody(context))
+        .imports(typeTable.getImports())
+        .build();
+  }
+
+  public SampleBodyView createSampleBody(SampleTransformerContext context) {
+    SampleConfig sampleConfig = context.getSampleConfig();
+    MethodInfo methodInfo = sampleConfig.methods().get(context.getMethodName());
+    SampleNamer sampleNamer = context.getNamer();
+    SampleTypeTable sampleTypeTable = context.getTypeTable();
+    SymbolTable symbolTable = new SymbolTable();
+
+    SampleBodyView.Builder sampleBodyView = SampleBodyView.newBuilder();
+    sampleBodyView.serviceVarName(symbolTable.getNewSymbol(sampleNamer.getServiceVarName()));
+    sampleBodyView.serviceTypeName(
+        sampleTypeTable.getAndSaveNicknameForServiceType(sampleConfig.apiTypeName()));
+    sampleBodyView.methodNameComponents(methodInfo.nameComponents());
+    sampleBodyView.requestVarName(symbolTable.getNewSymbol("request"));
+    sampleBodyView.requestTypeName(
+        sampleTypeTable.getAndSaveNicknameForRequestType(
+            sampleConfig.apiTypeName(), methodInfo.requestType().message().typeName()));
+
+    sampleBodyView.requestBodyVarName("");
+    sampleBodyView.requestBodyTypeName("");
+    sampleBodyView.resourceGetterName("");
+    sampleBodyView.resourceTypeName("");
+    sampleBodyView.isResourceMap(false);
+
+    if (methodInfo.isPageStreaming()) {
+      FieldInfo fieldInfo = methodInfo.pageStreamingResourceField();
+      if (fieldInfo == null) {
+        throw new IllegalArgumentException(
+            "method is page streaming, but the page streaming resource field is null.");
+      }
+      sampleBodyView.resourceGetterName(sampleNamer.getResourceGetterName(fieldInfo.name()));
+      String nickname = sampleTypeTable.getAndSaveNicknameFor(fieldInfo.type());
+      sampleBodyView.resourceTypeName(nickname);
+      // Rename the type from Map<K, V> to Map.Entry<K, V> to match what we
+      // expect in the for-loop.
+      if (fieldInfo.type().isMap()) {
+        sampleBodyView.resourceTypeName(sampleNamer.getMapEntryTypeFromMapType(nickname));
```

Comment:
It seems like getMapEntryTypeFromMapType should live in JavaSampleTypeNameConverter instead of SurfaceNamer, and it should take fieldInfo.type() as input.